examples/android_app/app/src/main/java/com/example/llamastackandroiddemo/ExampleLlamaStackLocalInference.kt

```diff
@@ -96,10 +135,22 @@ class ExampleLlamaStackLocalInference(
                 )
                 .build()
         )
-        response = result.asChatCompletionResponse().completionMessage().content().string();
+        response = result.asChatCompletionResponse().completionMessage().content().string().toString();
         tps =
             (result.asChatCompletionResponse()._additionalProperties()["tps"] as JsonNumber).value as Float
-        return response;
+
+        if (response == "") {
+            //Empty content as Llama Stack is returning a tool call
+            val toolCalls = result.asChatCompletionResponse().completionMessage().toolCalls()
+            return if (toolCalls.isNotEmpty()) {
+                functionDispatch(toolCalls, ctx)
+            } else {
+                "Empty tool calls and model response. File a bug"
```

Comment:
Assert instead?

Reply:
Assert will crash the app. With the method we have now it will print out the error so developer is still aware.